src/mmw/js/src/core/views.js

```diff
@@ -78,12 +79,55 @@ var HeaderView = Marionette.ItemView.extend({
         this.listenTo(this.appState, 'change', this.render);
     },
 
+    getNavClasses: function(isActive, isVisible) {
+        return (isActive ? 'active' : '') + (isVisible ? ' visible' : '');
+    },
+
+    isProjectsPage: function(currentActive) {
+        return currentActive === coreUtils.projectsPageTitle;
+    },
+
+    getSelectAreaNavClasses: function(currentActive) {
+        var isVisible = !this.isProjectsPage(currentActive),
+            isActive = currentActive === coreUtils.selectAreaPageTitle;
+        return this.getNavClasses(isActive, isVisible);
+    },
+
+    getAnalyzeNavClasses: function(currentActive, wasAnalyzeVisible, wasModelVisible) {
+        var isVisible = !this.isProjectsPage(currentActive) && (wasAnalyzeVisible || wasModelVisible),
+            isActive = currentActive === coreUtils.analyzePageTitle;
+        return this.getNavClasses(isActive, isVisible);
+    },
+
+    getModelNavClasses: function(currentActive) {
+        var modelPackageNames = _.pluck(settings.get('model_packages'), 'display_name'),
+            isCompare = currentActive === coreUtils.comparePageTitle,
+            isActive = _.contains(modelPackageNames, currentActive),
+            isVisible = isActive || isCompare;
+        return this.getNavClasses(isActive, isVisible);
+    },
+
+    getCompareNavClasses: function(currentActive, wasCompareVisible) {
```

Comment:
Nice job refactoring this! Picky naming thing: would it make sense to rename these methods to something like `makeCompareNavClasses` rather than `getCompareNavClasses` &c? I think what's happening is that we're creating a bundle of style switches to send to the template based on settings higher up in the hierarchy -- and that when those super-ordinate settings change, these work kind of like React functions to re-render the template?